torch/nn/modules/_functions.py

```diff
@@ -10,7 +10,7 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
         input = input.contiguous()
 
         count = torch.empty(1,
-                            dtype=running_mean.dtype,
+                            dtype=running_mean.dtype if running_mean is not None else torch.float32,
```

Comment:
since `count` is later catted with mean, better to use `mean` dtype for it instead of hardcoded float32. `mean` would be produced couple lines below, so moving `count` construction there should do the trick. Also, nit, use torch.full instead of torch.empty followed by `fill_`.